lib/src/controls/flyouts/menu.dart

```diff
@@ -0,0 +1,532 @@
+      width: size.width,
+      child: const Padding(
+        padding: EdgeInsetsDirectional.only(bottom: 5.0),
+        child: Divider(
+          style: DividerThemeData(horizontalMargin: EdgeInsets.zero),
+        ),
+      ),
+    );
+  }
+}
+
+enum SubItemShowBehavior {
+  /// Whether the sub-menu will be shown on item press
+  press,
+
+  /// Whether the sub-menu will be shown on item hover
+  ///
+  /// This is the default behavior.
+  hover,
+}
+
+/// Represents a menu item that displays a sub-menu in a [MenuFlyout].
+///
+/// ![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/menu-bar-submenu.png)
+///
+/// See also:
+///
+///  * [MenuFlyout], which displays a list of commands or options
+///  * [MenuFlyoutItem], a single item in the list of items
+///  * [MenuFlyoutSeparator], which represents a horizontal line that
+///    separates items in a [MenuFlyout].
+class MenuFlyoutSubItem extends MenuFlyoutItem {
+  /// Creates a menu flyout sub item
+  MenuFlyoutSubItem({
+    super.key,
+    super.leading,
+    required super.text,
+    super.trailing = const Icon(FluentIcons.chevron_right),
+    required this.items,
+    this.showBehavior = SubItemShowBehavior.hover,
+    this.showHoverDelay = const Duration(milliseconds: 600),
+  }) : super(onPressed: null);
+
+  GlobalKey<__MenuFlyoutSubItemState>? _key;
+
+  /// The colletion used to generate the content of the menu.
+  ///
+  /// {@macro fluent_ui.flyouts.menu.items}
+  final List<MenuFlyoutItemBase> items;
+
+  /// Represent which user action will show the sub-menu.
+  ///
+  /// Defaults to [SubItemShowBehavior.hover]
+  final SubItemShowBehavior showBehavior;
+
+  /// The sub-menu will be only shown after this delay
+  ///
+  /// Only applied if [showBehavior] is [SubItemShowBehavior.hover]
+  final Duration showHoverDelay;
+
+  @override
+  Widget build(BuildContext context) {
+    return _MenuFlyoutSubItem(key: _key, item: this, items: items);
+  }
+}
+
+class _MenuFlyoutSubItem extends StatefulWidget {
+  final MenuFlyoutSubItem item;
+  final List<MenuFlyoutItemBase> items;
+
+  const _MenuFlyoutSubItem({
+    Key? key,
+    required this.item,
+    required this.items,
+  }) : super(key: key);
+
+  @override
+  State<_MenuFlyoutSubItem> createState() => __MenuFlyoutSubItemState();
+}
+
+class __MenuFlyoutSubItemState extends State<_MenuFlyoutSubItem>
+    with SingleTickerProviderStateMixin {
+  /// The animation controller responsible for the animation of the flyout
+  ///
+  /// The duration is defined at build time
+  late final transitionController = AnimationController(vsync: this);
+
+  final menuKey = GlobalKey<_MenuFlyoutState>();
+
+  Timer? showTimer;
+
+  @override
+  void dispose() {
+    transitionController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final menuInfo = MenuInfoProvider.of(context);
+
+    transitionController.duration = menuInfo.widget.transitionDuration;
```

Comment:
this feels quite hacky, please rethink this logic to use a better solution, for example you could pass this parameter from the constructor itself and just use didUpdateWidget to update the duration